src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -209,74 +210,17 @@ public static async Task<JsonDocument> GetJsonCache(string uri, bool useCache =
-        /// </summary>
-        /// <param name="versionList"> list of version strings </param>
-        /// <returns> list of version strings, in sorted order </returns>
         public static IEnumerable<string> SortVersions(IEnumerable<string> versionList)
         {
-            // Scrub the version list
-            versionList = versionList.Select((v) =>
+            if (versionList == null || !versionList.Any())
             {
-                if (v.StartsWith("v", StringComparison.InvariantCultureIgnoreCase))
-                {
-                    return v.Substring(1).Trim();
-                }
-                else
-                {
-                    return v.Trim();
-                }
-            });
-
-            // Attempt to sort using different methods
-            List<Func<string, object>> methods = new List<Func<string, object>>
-            {
-                (s) => new Version(s),
-                (s) => new SemVer.Version(s, loose: true),
-                (s) => s
-            };
-
-            // Iterate through each method we defined above.
-            foreach (var method in methods)
-            {
-                var objList = new List<object>();
-                try
-                {
-                    foreach (var version in versionList)
-                    {
-                        var verResult = method(version);
-                        // Make sure the method doesn't mangle the version This is due to System.Version
-                        // normalizalizing "0.01" to "0.1".
-                        if (verResult != null && (verResult.ToString() ?? string.Empty).Equals(version))
-                        {
-                            objList.Add(verResult);
-                        }
-                        else
-                        {
-                            Logger.Debug("Mangled version [{0}] => [{1}]", version, verResult);
-                        }
-                    }
-                    objList.Sort();  // Sort using the built-in sort, delegating to the type's comparator
-                }
-                catch (Exception)
-                {
-                    objList = null;
-                }
-
-                // If we have a successful result (right size), then we should be good.
-                if (objList != null && objList.Count() == versionList.Count())
-                {
-                    return objList.Select(o => o.ToString() ?? string.Empty);
-                }
+                return Array.Empty<string>();
             }
 
-            // Fallback, leaving it alone
-            if (Logger.IsDebugEnabled)  // expensive string join, avoid unless necessary
-            {
-                Logger.Debug("List is not sortable, returning as-is: {0}", string.Join(", ", versionList));
-            }
-            return versionList;
+            // Split Versions
+            var versionPartsList = versionList.Select(s => VersionComparer.Parse(s)).ToList();
```

Comment:
I don't think you need this.  I tested and the default sort seems to understand version numbers. You should just be able to sort the strings without all this manipulation.

```csharp
> var str1 = "12.1.1";
> var str2 = "2.1.1";
> var list = new List<string>();
> list.Add(str2);
> list.Add(str1);
> list
List<string>(2) { "2.1.1", "12.1.1" }
> list.Sort()
> list
List<string>(2) { "12.1.1", "2.1.1" }
```